File: backend/tasks/agents/analyst/feed_summary_agent.py
"""FeedSummaryAgent - Simple no-tools agent that generates a short feed summary."""

import json
import time
from tasks import register_agent
from tasks.llm.config import CODING_MODEL, ONBOARDING_MODEL
from tasks.llm.models import LlmSettings
from .agent import AnalystAgent
from .prompt_loader import get_prompt


@register_agent
class FeedSummaryAgent(AnalystAgent):
    """
    Minimal agent that takes recent file metadata and produces a 2-3 sentence
    summary for the home feed. No tools — single LLM call, returns text.
    """

    def _get_system_message(self) -> dict:
        content = get_prompt('feed_summary.system', agent_name=self.agent_name)
        return {"role": "system", "content": content}

    def _get_user_message(self) -> dict:
        app_state_str = json.dumps(self.app_state, indent=2) if self.app_state else "null"
        print(f"[FeedSummaryAgent] app_state:\n{app_state_str}")
        content = get_prompt(
            'feed_summary.user',
            app_state=app_state_str,
            current_date=time.strftime("%Y-%m-%d"),
        )
        return {"role": "user", "content": content}

    def _get_llm_settings(self) -> LlmSettings:
        return LlmSettings(
            model=ONBOARDING_MODEL,
            response_format={"type": "text"},
            tool_choice="none",
            include_web_search=False,
        )

    def _get_available_tools(self):
        return []

File: frontend/agents/feed-summary/feed-summary-agent.ts
// FeedSummaryAgent.
//
// A leaf agent with NO tools: a single LLM call that turns recent-file app
// state into a 2-3 sentence home-feed summary. Renders the `feed_summary.system`
// / `feed_summary.user` prompts (in `orchestrator/prompts/prompts.yaml`). Runs
// headless via `runFeedSummaryV2`.
import 'server-only';
import { todayISO } from '@/lib/utils/today';
import { Type } from 'typebox';
import type { Tool, TextContent, ImageContent } from '@/orchestrator/llm';
import { renderPrompt } from '@/orchestrator/prompts';
import { registerFauxProvider } from '@/orchestrator/llm/testing';
import { RemoteAnalystAgent } from '@/agents/analyst/analyst-agent';
import { getAgentModelOrTestFallback } from '@/agents/analyst/model-config';
import { appStateForLlm, type AppState } from '@/lib/appState';

export const fauxRegistration = registerFauxProvider({
  api: 'faux-feed-summary-api',
  provider: 'faux-feed-summary',
  models: [{ id: 'stub-feed-summary' }],
});
const FAUX_MODEL = fauxRegistration.getModel();

const FeedSummaryAgentParams = Type.Object({
  userMessage: Type.String(),
});

export class FeedSummaryAgent extends RemoteAnalystAgent {
  static readonly schema: Tool<typeof FeedSummaryAgentParams> = {
    name: 'FeedSummaryAgent',
    description: 'Generates a short home-feed summary from recent-file app state. No tools, single LLM call.',
    parameters: FeedSummaryAgentParams,
  };
  // No tools — one LLM turn, returns text.
  static override readonly tools: Tool<typeof FeedSummaryAgentParams>[] = [];
  static override model = getAgentModelOrTestFallback(FAUX_MODEL);

  protected override getSystemPrompt(): string {
    return renderPrompt('feed_summary.system', {
      agent_name: this.context.agentName ?? 'MinusX',
    });
  }

  protected override buildUserContent(): (TextContent | ImageContent)[] {
    const appStateJson =
      this.context.appState !== undefined ? JSON.stringify(appStateForLlm(this.context.appState as AppState), null, 2) : 'null';
    const text = renderPrompt('feed_summary.user', {
      app_state: appStateJson,
      current_date: todayISO(),
    });
    return [{ type: 'text', text }];
  }
}
